src/wallet-account.ts

```diff
@@ -222,14 +244,15 @@ export class WalletConnection {
         const accountId = currentUrl.searchParams.get('account_id') || '';
         // TODO: Handle errors during login
         if (accountId) {
-            this._authData = {
+            const authData = {
                 accountId,
                 allKeys
             };
-            window.localStorage.setItem(this._authDataKey, JSON.stringify(this._authData));
+            window.localStorage.setItem(this._authDataKey, JSON.stringify(authData));
             if (publicKey) {
                 await this._moveKeyFromTempToPermanent(accountId, publicKey);
             }
+            this._authData = authData;
```

Comment:
Any reason in particular we need to make these changes?

Reply:
The intention here is to only change `this.authData` on the instance once sign in is actually finished and the localStorage changes have completed successfully. Agreed ✅

Reply:
The intention here is to only modify `_authData` on the instance once sign in has _actually_ been completed successfully and the required localStorage changes have been made ✅